### PR TITLE
Added support for bulk deletes with test

### PR DESCRIPTION
**Pull request by agearhart**
Please let me know if you have any issues with this code.  :)

### CODE REVIEW

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala [hunk 1]
```diff
@@ -248,6 +253,7 @@ object RestlasticSearchClient {
     case class BulkItem(_index: String, _type: String, _id: String, status: Int, error: Option[String]) {
       def created = status > 200 && status < 299 && !alreadyExists
       def alreadyExists = error.exists(_.contains("DocumentAlreadyExists"))
+      def deleted = status == 200
```
Comment (rcoh) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala]: This is a confusing name because the same object is returned by the create action. Can you rename this to `success` and have it be >= 200 <= 299?

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala [hunk 2]
```diff
@@ -168,6 +168,52 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
         res.jsonStr should not include("doc5")
       }
     }
+    
+    "Support bulk deletion" in {
```
Comment (rcoh) [inline elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala]: Either add deletion to the previous test, or do the setup without checking the result to avoid duplicate tests.

**Review by rcoh — CHANGES_REQUESTED** [review]
Just a couple fixes

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala [hunk 3]
```diff
@@ -135,6 +135,11 @@ class RestlasticSearchClient(endpointProvider: EndpointProvider, signer: Option[
     val bulkOperation = Bulk(documents.map(BulkOperation(update, Some(index -> tpe), _)))
     bulkIndex(bulkOperation)
   }
+  
+  def bulkDelete(index: Index, tpe: Type, documents: Seq[Document]): Future[Seq[BulkItem]] = {
```
Comment (CCheSumo) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala:139]: Shouldn't we support delete by document ids instead of requiring `documents: Seq[Document]` to be passed in?

**Review by agearhart — COMMENTED** [review]
I've renamed BulkItem.deleted to BulkItem.success and combined the tests as suggested.  Please let me know if this is what was intended.

**Comment by maryammzadeh** [comment]
This is awesome! I was hoping there would be support for bulk deletes soon! Thanks!

**Review by rcoh — APPROVED** [review]

**Review by CCheSumo — COMMENTED** [review]
@pnzrdrgoon one more comment

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala (hunk 3 above)
Comment (agearhart) [reply elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala:139]: We can totally do that.  I was just keeping with conventions.  I'll prop that change soon.
